src/analog/m2kanalogin_impl.cpp

```diff
@@ -605,37 +582,82 @@ double M2kAnalogInImpl::getValueForRange(M2K_RANGE range)
 	}
 }
 
-struct IIO_OBJECTS M2kAnalogInImpl::getIioObjects()
+struct libm2k::IIO_OBJECTS M2kAnalogInImpl::getIioObjects()
 {
-	return DeviceIn::getIioObjects();
+	return m_m2k_adc->getIioObjects();
 }
 
 void M2kAnalogInImpl::cancelBuffer()
 {
-	DeviceIn::cancelBuffer();
+	m_m2k_adc->cancelBuffer();
 }
 
 void M2kAnalogInImpl::setKernelBuffersCount(unsigned int count)
 {
-	DeviceGeneric::setKernelBuffersCount(count);
+	m_m2k_adc->setKernelBuffersCount(count);
 }
 
 std::vector<double> M2kAnalogInImpl::getAvailableSampleRates()
 {
-	return DeviceGeneric::getAvailableSampleRates();
+	return m_m2k_adc->getAvailableSampleRates();
 }
 
 std::string M2kAnalogInImpl::getName()
 {
-	return DeviceGeneric::getName();
+	return m_m2k_adc->getName();
 }
 
 void M2kAnalogInImpl::convertChannelHostFormat(unsigned int chn_idx, int16_t *avg, int16_t *src)
 {
-	DeviceGeneric::convertChannelHostFormat(chn_idx, avg, src, false);
+	m_m2k_adc->convertChannelHostFormat(chn_idx, avg, src, false);
 }
 
 void M2kAnalogInImpl::convertChannelHostFormat(unsigned int chn_idx, double *avg, int16_t *src)
 {
-	DeviceGeneric::convertChannelHostFormat(chn_idx, avg, src, false);
+	m_m2k_adc->convertChannelHostFormat(chn_idx, avg, src, false);
+}
+
+int M2kAnalogInImpl::convertVoltsToRawVerticalOffset(ANALOG_IN_CHANNEL channel, double vertOffset)
+{
+	double gain = 1.3;
+	double vref = 1.2;
+	double hw_range_gain = getValueForRange(m_input_range.at(channel));
+	int raw_vert_offset = (int) (vertOffset * (1u << 12u) * hw_range_gain * gain / 2.693 / vref);
+	return raw_vert_offset;
```

Comment:
I think you can directly return the computation here without the need for a temporary variable `raw_vert_offset`.
Also c++ style casts should be preferred instead of the C ones.

Reply:
The modification was made.